File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/json/UriTypeAdapter.java
package com.elveselimoski.starterapp.json;

import android.net.Uri;
import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;

import java.io.IOException;

public class UriTypeAdapter extends TypeAdapter<Uri> {

    public void write(JsonWriter out, Uri value) throws IOException {
        out.value(value.toString());
    }

    public Uri read(JsonReader in) throws IOException {
        String url = in.nextString();
        return Uri.parse(url);
    }
}

File: StarterApp/app/src/main/java/com/elveselimoski/starterapp/json/UriTypeAdapter.kt
package com.elveselimoski.starterapp.json

import android.net.Uri
import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter

import java.io.IOException

class UriTypeAdapter : TypeAdapter<Uri>() {
    @Throws(IOException::class)
    override fun write(out: JsonWriter, value: Uri) {
        out.value(value.toString())
    }

    @Throws(IOException::class)
    override fun read(`in`: JsonReader): Uri {
        val url = `in`.nextString()
        return Uri.parse(url)
    }
}
